sway-lib-std/src/vec.sw

```diff
@@ -6,16 +6,28 @@ use ::intrinsics::size_of;
 use ::mem::{copy, read, write};
 use ::option::Option;
 
+fn raw_ptr_from_u64(ptr: u64) -> raw_ptr {
+    asm(r1: ptr) {
+        r1: raw_ptr
+    }
+}
+
+fn raw_ptr_into_u64(ptr: raw_ptr) -> u64 {
+    asm(r1: ptr) {
+        r1: u64
+    }
+}
+
```

Comment:
I dislike these 😢 but I understand the need for them. Maybe we should, in fact, have `alloc` return a `raw_ptr` and [follow what Andrew did](https://github.com/FuelLabs/sway/pull/2699/files) (the whole thing with `fn offset_ptr()`. I know that in that case, some standard library tests will fail because of our check, but that's okay because we can modify them to return a `u64` via an `asm` block that converts that `raw_ptr` into a `u64` (in the tests themselves). What do you think?